### DISCUSSION TITLE
How do I specify torch::nn::Upsample scale_factor?

### DISCUSSION THREAD

**Discussion by rfryeSigma** [idaholab/moose#27620] [Q&A General] [answered]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

I am tryting to specify an Upsample module in a `torch::nn::Sequential` argument list. The documentation in 
`moose/framework/contrib/libtorch/include/torch/csrc/api/include/torch/nn/options/upsampling.h`
gives the example
`/// model(UpsampleOptions().scale_factor({3}).mode(torch::kLinear).align_corners(false));`
but I don't know how to apply that information.
I have tried this line
`  torch::nn::Upsample(),
`
and it compiles but gets the runtime error "either size or scale_factor should be defined".
I want to specify a scale_factor of 2. These are some of the ways that I have tried to implement this:
```
      torch::nn::Upsample(nullptr, 2.0),
      torch::nn::Upsample(void, 2.0),
      torch::nn::Upsample(2.0),
      torch::nn::Upsample(scale_factor{2.0}),
      torch::nn::Upsample(scale_factor={2}),
      torch::nn::Upsample(scale_factor({2})),
      torch::nn::Upsample(torch::nn::UpsampleOptions().scale_factor({2})),
      torch::nn::Upsample(torch::nn::UpsampleOptions(scale_factor({2}))),

```
but none of them compile.

**Comment by GiudGiud** [comment]
Hello

seems more of a torch question than a moose one.
@grmnptr or @somu15  might know still

Guillaume

**Reply by rfryeSigma** [reply, marked as answer]
I found the answer in [pytorch cppdocs](https://pytorch.org/cppdocs/api/structtorch_1_1nn_1_1_upsample_options.html#structtorch_1_1nn_1_1_upsample_options)

`torch::nn::Upsample(torch::nn::UpsampleOptions().scale_factor(std::vector<double>({2}))),`
